pkgs/development/libraries/imgui/default.nix

```diff
@@ -1,8 +1,77 @@
-{ stdenv, lib, fetchFromGitHub }:
+{
+  stdenv,
+  lib,
+  applyPatches,
+  callPackage,
+  cmake,
+  fetchFromGitHub,
+  fetchpatch,
+  glfw,
+  libGL,
+  darwin,
+  vcpkg,
+  SDL2,
+
+  # The intent is to mirror vcpkg's flags[^1],
+  # but we limit our attention to Linux and glfw3 for now
+  # [^1]: https://github.com/microsoft/vcpkg/blob/095ee06e7f60dceef7d713e3f8b1c2eb10d650d7/ports/imgui/CMakeLists.txt#L33-L108
+  withGLLoader ? !libGL.meta.unsupported,
+  withFeatures ?
+    lib.optionals (!glfw.meta.unsupported) [ "glfw" ]
+    ++ lib.optionals (!SDL2.meta.unsupported) [ "sdl2" ]
+    ++ lib.optionals (!stdenv.isDarwin) [ "opengl3" ]
+    ++ lib.optionals stdenv.isDarwin [ "metal" ]
+    ++ lib.optionals stdenv.hostPlatform.isAndroid [ "android" ],
```

Comment:
I'm quite unhappy about making this a list instead of individual `bool` attributes, but OTOH there's the issue of making a consistent mapping from the nix expression's arguments to vcpkg/cmake options and I'd rather not over-engineer it